### PR TITLE
[ENH] Modified `BaseDeepClusterer`

### PR DESCRIPTION
**Pull request by AnonymousCodes911**
fixes: #1170 
added `KShape` and `KMedoids` to `BaseDeepClusterer`

### CODE REVIEW

File: aeon/clustering/deep_learning/base.py [hunk 1]
```diff
@@ -116,6 +124,16 @@ def _fit_clustering(self, X):
                 self.clusterer = TimeSeriesKMeans(
                     n_clusters=self.n_clusters, **clustering_params_
                 )
+        elif self.clustering_algorithm == "kshape":
+            self.clusterer = TimeSeriesKShapes(
+                n_clusters=self.n_clusters, **clustering_params_
+            )
```
Comment (MatthewMiddlehurst) [inline aeon/clustering/deep_learning/base.py:130]: The new `elif` and the above code should be similar, if the `if len(clustering_params_.keys()) == 0:` line is not needed it should be removed.

File: aeon/clustering/deep_learning/base.py [hunk 2]
```diff
@@ -116,6 +124,16 @@ def _fit_clustering(self, X):
...
+        elif self.clustering_algorithm == "kmedoids":
+            self.clusterer = TimeSeriesKMedoids(
+                n_clusters=self.n_clusters, **clustering_params_
+            )
+        else:
+            raise ValueError(f"Invalid algorithm: {self.clustering_algorithm}")
```
Comment (MatthewMiddlehurst) [inline aeon/clustering/deep_learning/base.py:136]: could you add the parameter name in the error somewhere, i.e. "Invalid input for clustering_algorithm"

File: aeon/clustering/deep_learning/base.py [hunk 3]
```diff
@@ -15,8 +17,13 @@ class BaseDeepClusterer(BaseClusterer, ABC):
     ----------
     n_clusters : int, default=None
         Number of clusters for the deep learning model.
-    clustering_algorithm : str, default="kmeans"
+    clustering_algorithm : str, {'kmeans', 'kshape', 'kmedoids'},
+        default="kmeans"
         The clustering algorithm used in the latent space.
+        Options include:
+        'kmeans' for K-means clustering,
```
Comment (MatthewMiddlehurst) [inline aeon/clustering/deep_learning/base.py:24]: `K-means`  to `KMeans` if we want to keep the style consistent

**Review by MatthewMiddlehurst — COMMENTED** [review]
Could you add a test in `test_deep_clusterer_base.py` for your new options? The input can be a parameter for the test.

File: aeon/clustering/deep_learning/base.py (hunk 1 above)
Comment (AnonymousCodes911) [reply aeon/clustering/deep_learning/base.py:126]: cleaned the code!

**Comment by MatthewMiddlehurst** [comment]
The test should be new, not an edit the the current test.

**Review by hadifawaz1999 — APPROVED** [review]
lgtm

**Pull request merged**